src/main/kotlin/com/sourcegraph/cody/context/RemoteRepoInsight.kt

```diff
@@ -0,0 +1,394 @@
+import com.intellij.extapi.psi.PsiFileBase
+import com.intellij.lang.ASTNode
+import com.intellij.lang.Language
+import com.intellij.lang.ParserDefinition
+import com.intellij.lang.PsiParser
+import com.intellij.lang.annotation.AnnotationHolder
+import com.intellij.lang.annotation.Annotator
+import com.intellij.lang.annotation.HighlightSeverity
+import com.intellij.lexer.Lexer
+import com.intellij.lexer.LexerPosition
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.fileTypes.FileType
+import com.intellij.openapi.fileTypes.LanguageFileType
+import com.intellij.openapi.progress.*
+import com.intellij.openapi.project.DumbAware
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.util.TextRange
+import com.intellij.patterns.PlatformPatterns
+import com.intellij.psi.FileViewProvider
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiFile
+import com.intellij.psi.tree.IElementType
+import com.intellij.psi.tree.IFileElementType
+import com.intellij.psi.tree.TokenSet
+import com.intellij.psi.util.elementType
+import com.intellij.refactoring.suggested.endOffset
+import com.intellij.refactoring.suggested.startOffset
+import com.intellij.util.ProcessingContext
+import com.jetbrains.rd.util.getThrowableText
+import com.sourcegraph.Icons
+import com.sourcegraph.cody.context.ui.MAX_REMOTE_REPOSITORY_COUNT
+import com.sourcegraph.common.CodyBundle
+import com.sourcegraph.common.CodyBundle.fmt
+import javax.swing.Icon
+import org.jetbrains.annotations.NonNls
+
+data class RemoteRepo(val name: String) {
+  val displayName: String
+    get() = name.substring(name.indexOf('/') + 1) // Note, works for names without / => full name.
+
+  val icon: Icon?
+    get() =
+        when {
+          name.startsWith("github.com/") -> Icons.RepoHostGitHub
+          name.startsWith("gitlab.com/") -> Icons.RepoHostGitlab
+          name.startsWith("bitbucket.org/") -> Icons.RepoHostBitbucket
+          else -> Icons.RepoHostGeneric
+        }
+}
+
+val RemoteRepoLanguage = object : Language("SourcegraphRemoteRepoList") {}
+
+class RemoteRepoFileType : LanguageFileType(RemoteRepoLanguage) {
+  companion object {
+    @JvmStatic val INSTANCE = RemoteRepoFileType()
+  }
+
+  override fun getName(): String {
+    return "SourcegraphRemoteRepoListFile"
+  }
+
+  override fun getDescription(): String {
+    return "A list of Sourcegraph repository indexes"
+  }
+
+  override fun getDefaultExtension(): String {
+    return ""
+  }
+
+  override fun getIcon(): Icon? {
+    return null
+  }
+}
+
+class RemoteRepoTokenType(debugName: @NonNls String) : IElementType(debugName, RemoteRepoLanguage) {
+  override fun toString(): String {
+    return "RemoteRepoTokenType." + super.toString()
+  }
+
+  companion object {
+    val REPO = RemoteRepoTokenType("REPO")
+    val SEPARATOR = RemoteRepoTokenType("SEPARATOR")
+    val EOF = RemoteRepoTokenType("EOF")
+  }
+}
+
+class RemoteRepoFile(viewProvider: FileViewProvider?) :
```

Comment:
If that class does not make sense without `viewProvider` can we please change it to be `class RemoteRepoFile(viewProvider: FileViewProvider)`?